dictionary/MessageFormatDictionary.js

```diff
@@ -80,22 +80,54 @@ export default class MessageFormatDictionary extends Dictionary {
 	 *        Defaults to an empty plain object.
 	 */
 	get(key, parameters = {}) {
+		let scope = this._getScope(key);
+
+		if (scope === false) {
+			throw new GenericError(
+				`ima.dictionary.MessageFormatDictionary.get: The ` +
+				`localization phrase '${key}' does not exists`,
+				{ key, parameters }
+			);
+		}
+
+		return scope(parameters);
+	}
+
+	/**
+	 * @inheritdoc
+	 * @param {string} key The key identifying the localization phrase. The key
+	 *        consists of at least two parts separated by dots. The first part
+	 *        denotes the name of the source JSON localization file, while the
+	 *        rest denote a field path within the localization object within
+	 *        the given localization file.
+	 */
+	has(key) {
+		let scope = this._getScope(key);
+		return (scope === false) ? false : true;
+	}
+
+	/**
+	 * @inheritdoc
+	 * @param {string} key The key identifying the localization phrase. The key
+	 *        consists of at least two parts separated by dots. The first part
+	 *        denotes the name of the source JSON localization file, while the
+	 *        rest denote a field path within the localization object within
+	 *        the given localization file.
+	 */
+	_getScope(key) {
 		let path = key.split('.');
 		let scope = this._dictionary;
 
 		for (let scopeKey of path) {
 			if (!scope[scopeKey]) {
-				throw new GenericError(
-					`ima.dictionary.MessageFormatDictionary.get: The ` +
-					`localization phrase '${key}' does not exists`,
-					{ key, parameters }
-				);
+				return false;
 			}
 
 			scope = scope[scopeKey];
 		}
 
-		return scope(parameters);
+		// only if scope is function, the key is useable
+		return (typeof scope === "function") ? scope : false;
```

Comment:
I would prefer `return scope` here since this method is meant only to retrieve a scope (hence its name `_getScope`). This method may become more useful in the future, so I would prefer to keep it more general-purpose.

Your objection is correct (thank you for that!), but you propose to solve it in the wrong place. To check whether the `key` passed to the `has()` method is referring to a localization phrase, I propose adding the following check to the `has()` method:

```javascript
if (!/^[^.]+\.[^.]+$/.test(key)) {
	throw new Error(
		`The provided key (${key}) is not a valid localization phrase key, ` +
		`expecting a "file name.identifier" notation`.
	);
}
```